tests/dev.rs

```diff
@@ -127,6 +127,5 @@ pub fn test_roundtrip() {
 
     println!("...read 2 successfull");
 
-    assert!(!image.contains_nan_pixels() && !image2.contains_nan_pixels());
-    assert_eq!(image, image2);
+    assert!(image.similar_to_lossy(&image2, 0.05));
```

Comment:
Juste to be sure: Are you using `similar_to_lossy()` also for lossless compression?

Reply:
It's used everywhere, where an image is compared in the tests. I know this because I replaced all usages of `ContainsNaN` which was used in every test previously. These tests run for all exr images in the repository, including some B44 images. We can always add more images to the test folder to increase coverage.

Reply:
However, I assume most images will not change. Because the image files in the repository have already been compressed with lossy methods, opening the images and compressing them again will likely not loose any additional data. We still have the tests from your PR though, where an image will be opened, then the compression method will be set to B44, and then it will be compressed. So I see no problem considering this.

Reply:
I was asking because I think it can be a good thing to keep a strong equal comparison for non destructive compression.

Reply:
I think you're right, nice. Does this mean that f32 data will be completely uncompressed in the file?! 

Also, why is it necessary to copy these channels twice (in to tmp, tmp to out)? We could save one copy and some memory there (just an idea for later, not now).

Reply:
Yes, I thought of that, but no design I came up with satisfied me. I think I have a better idea now, let me try

Reply:
Pushed a new design implementing this idea